py/server/deephaven/parquet.py

```diff
@@ -113,12 +159,36 @@ def read(
             is_legacy_parquet=is_legacy_parquet,
             is_refreshing=is_refreshing,
             for_read=True,
+            force_build=True,
         )
-
-        if read_instructions:
-            return Table(j_table=_JParquetTools.readTable(path, read_instructions))
+        j_table_definition = _j_table_definition(table_definition)
+        if j_table_definition is not None:
+            if not type:
+                raise DHError("Must provide type when table_definition is set")
+            if type == ParquetType.SINGLE:
+                j_table = _JParquetTools.readSingleTable(_JFile(path), read_instructions, j_table_definition)
+            elif type == ParquetType.FLAT_PARTITIONED:
+                j_table = _JParquetTools.readFlatPartitionedTable(_JFile(path), read_instructions, j_table_definition)
+            elif type == ParquetType.KV_PARTITIONED:
+                j_table = _JParquetTools.readKeyValuePartitionedTable(_JFile(path), read_instructions, j_table_definition)
+            elif type == ParquetType.METADATA_PARTITIONED:
+                raise DHError(f"{ParquetType.METADATA_PARTITIONED} with table_definition not currently supported")
```

Comment:
Do you want the f-string here?

Reply:
Just cleanly support refactoring if `METADATA_PARTITIONED` was renamed.

Reply:
Sounds good.